jenkins/opensearch/perf-test.jenkinsfile

```diff
@@ -0,0 +1,153 @@
+            trim: true
+        )
+        string(
+            defaultValue: 'nyc_taxis',
+            name: 'TEST_WORKLOAD',
+            description: 'The workload name from OpenSearch Benchmark Workloads for Mensor (internal client).',
+            trim: true
+        )
+        string(
+            defaultValue: '1',
+            name: 'TEST_ITERATIONS',
+            description: 'Number of times to run a workload for Mensor (internal client).',
+            trim: true
+        )
+        string(
+            defaultValue: '0',
+            name: 'WARMUP_ITERATIONS',
+            description: 'Number of times to run a workload before collecting data for Mensor (internal client).',
+            trim: true
+        )
+    }
+
+    stages {
+        stage('validate-parameters') {
+            steps {
+                script {
+                    if (BUNDLE_MANIFEST_URL == '') {
+                        currentBuild.result = 'ABORTED'
+                        error("Performance Tests failed to start. Missing parameter: BUNDLE_MANIFEST_URL.")
+                    }
+                    if (GITHUB_TOKEN == '') {
+                        currentBuild.result = 'ABORTED'
+                        error("Performance Tests failed to start. Missing parameter: GITHUB_TOKEN.")
+                    }
+                }
+            }
+        }
+        stage('perf-test') {
+            agent {
+                docker {
+                    label AGENT_X64
+                    image AGENT_IMAGE
+                    alwaysPull true
+                }
+            }
+            steps {
+                script {
+                    def bundleManifestObj = downloadBuildManifest(
+                        url: BUNDLE_MANIFEST_URL,
+                        path: BUNDLE_MANIFEST
+                    )
+                    String buildId = bundleManifestObj.getArtifactBuildId()
+                    env.BUILD_ID = buildId
+                    env.HAS_SECURITY = bundleManifestObj.components.containsKey("security")
+                    echo "BUNDLE_MANIFEST: ${BUNDLE_MANIFEST}"
+                    echo "BUILD_ID: ${BUILD_ID}"
+                    echo "Security: ${HAS_SECURITY}"
+
+                    lib.jenkins.Messages.new(this).add('perf-test', "[Test]Executing performance tests for #${BUILD_ID}")
+
+                    runPerfTestScript(bundleManifest: "${BUNDLE_MANIFEST}",
+                        buildId: "${BUILD_ID}",
+                        security: false,
+                        workload: TEST_WORKLOAD,
+                        testIterations: TEST_ITERATIONS,
+                        warmupIterations: WARMUP_ITERATIONS)
+
+                    lib.jenkins.Messages.new(this).add('perf-test', "[Test]Performance tests without security for ${BUILD_ID} completed")
+
+                    if(env.HAS_SECURITY) {
+                         runPerfTestScript(bundleManifest: "${BUNDLE_MANIFEST}",
```

Comment:
Given the amount of time each run takes, can we run this in parallel?

Reply:
I will parallelize this going forward. We still need a few test runs to iron out all the time outliers and other execution issues, if any.

Reply:
Do we plan to do it in this PR?

Reply:
At this point I'd recommend getting this PR in, and then cut a separate PR.  I'd much rather get in a solid base and iterate than hold for a big PR with everything in it.  Remember: Progress, not perfection! :)

Reply:
Sure, let's parallelize the job in a different PR. @kotwanikunal - Can you open an issue to parallelize this job?

Let's fix the failing tests and address the other review comments fixing the new tests that are added before we get this in.

Thanks!

Reply:
@kotwanikunal - Can you please post the issue link here after creating it so it can be connected and will be easy to trace? Thank you!

Reply:
great!